packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -39,7 +42,17 @@ export default class ConfigParser {
         private _moduleRequireService: ModuleRequireService = new RequireLibrary()
     ) {}
 
-    autoCompile() {
+    async autoCompile() {
+        /**
+         * Checks tsconfig.json path, throws error if it doesn't exist
+         */
+        if (this._config.autoCompileOpts?.tsNodeOpts?.project) {
+            const tsconfigPath = path.resolve(this._config.autoCompileOpts.tsNodeOpts.project)
+            if (!canAccess(tsconfigPath)) {
+                throw new Error(`Provided tsconfig file path '${tsconfigPath}' in wdio config is incorrect. Is it correctly set in wdio config ?`)
+            }
+        }
```

Comment:
Given that everything is sync here, I'ld suggest to move this into the [`loadTypeScriptCompiler`](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-config/src/utils.ts#L113-L133) function.

Reply:
Sure!